File: 0001_0100/0011/0011.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    k, n = (int(x) for x in f.read().split())
    a = [1, 1]
    for i in range(2, n + 1):
        a.append(a[-1] * 2)
        if (i > k):
            a[i] -= a[i - k - 1]
    q.write(str(a[n]))

File: 0001_0100/0011/0011.go
package main

import (
	"fmt"
	"math/big"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var k, n int
	fmt.Fscan(f, &k, &n)
	p := []*big.Int{big.NewInt(1), big.NewInt(1)}
	for i := 2; i <= n; i++ {
		p = append(p, new(big.Int).Mul(big.NewInt(2), p[i-1]))
		if i > k {
			p[i] = new(big.Int).Sub(p[i], p[i-k-1])
		}
	}
	fmt.Fprint(q, p[n])
}
